docs/source/format/CanonicalExtensions.rst

```diff
@@ -251,6 +251,27 @@ Variable shape tensor
    Values inside each **data** tensor element are stored in row-major/C-contiguous
    order according to the corresponding **shape**.
 
+.. _json_extension:
+
+JSON
+====
+
+* Extension name: ``arrow.json``.
+
+* The storage type of this extension is ``StringArray`` or
+  or ``LargeStringArray`` or ``StringViewArray``.
```

Comment:
One further nit: "StringArray" is not really generic terminology that we use in the format specs (it's the name of the array/vector in Arrow C++ and pyarrow, but not necessarily in other implementations). So I think you can leave out the `Array` part in each of those three types. 

(strictly speaking the spec actually uses Utf8 / LargeUtf8 / Utf8View, but I think in many places we do use "String" instead)

Reply:
Let's use the same wording as in the spec, then?

Reply:
@jorisvandenbossche like so?
```suggestion
* The storage type of this extension is ``String`` or
  or ``LargeString`` or ``StringView``.
```